### DISCUSSION TITLE
Placing Images on a PDF single page

### DISCUSSION THREAD

**Discussion by eccare** [ImageMagick/ImageMagick#2443] [Help]
I recently upgraded operating systems from Centos 5.10 to Centos 7.8, and the conversion broke software that worked on the previous version.   I am trying to put two signatures  and one logo on a created pdf form.  The method I used previuosly was to  use the following steps :

#1. I create a pdf file from an fdf data file with pdftk
/usr/bin/pdftk tdi73-1.pdf fill_form wrk-99-800893yea.fdf output pg1-99-800893yea.pdf

#2. I put a signature on a previously created (white page Canvas
/usr/bin/composite -geometry +180+1019 sig.p0101.png canvas.pdf 73--p.pdf

#3. I put a second signature on the pdf file created above 73.p.pdf
/usr/bin/composite -geometry +160+949 dsig.jpg 73-p.pdf 73.t.pdf

This process created two signatures in the proper place on a pdf file, but after the conversion this process puts the second signature in the proper place but the first signature is raised to the middle of the page and multipled three times across a single line.  This did not happen berfore. 

There are other procedures done before completion, but this is where the abnormalities begin

Your help would be appreciated !!!

Greg Ennis

**Comment by fmw42** [comment]
It is likely a change in version of ImageMagick, or you do not have Ghostscript installed to read PDF files or your new OS has restricted the use of PDF/PS etc files, which requires a modification of the policy.xml file. Please identify your two IM versions and their delegate libraries and also tell us in what way it does not work.

**Comment by snibgo** [comment]
Please also show your files.

In one command, you have "73--p.pdf" but another has "73-p.pdf".

What you are doing isn't wise. ImageMagick is a raster image processor. When it reads a PDF, it rasterizes each page, even if that page happens to contain one or more raster images. IM won't "put a signature" on a PDF page.
